### PR TITLE
Fix sending contact with IBAN

### PR DESCRIPTION
**Pull request by clenrock**
- Relevant Issues: -
- Related Pull Requests: -

* * *

## What

This Pull Request adds to the repository the right way to convert a profile into a contact for NFC, QR and the VisualVerification.

## Why

This Pull Request is needed because the IBAN should be also send through bluetooth.

## How

This feature can be viewed/tested within the project by sending a contact.

## Notes

None

### CODE REVIEW

File: app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeActivity.java [hunk 1]
```diff
@@ -112,7 +112,7 @@ public void onActivityResult(final int requestCode, final int resultCode, final
             LOGGER.error("No scan result in QR Scanner");
         } else if (resultCode == RESULT_OK) {
             final String result = scanResult.getContents();
-            LOGGER.info("Adding new contact to database");
+            LOGGER.info("QR code scanned: " + result);
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/activities/QRExchangeActivity.java]: Use extra params and {}

File: app/src/main/java/com/nervousfish/nervousfish/data_objects/Ed25519Key.java [hunk 2]
```diff
@@ -113,7 +113,7 @@ public static EdDSAPrivateKey getPrivateKeyOf(final Ed25519Key key) {
      */
     @Override
     public String getKey() {
-        return this.key;
+        return this.key + "#" + Arrays.toString(this.seed);
```
Comment (TheBonheurs) [inline app/src/main/java/com/nervousfish/nervousfish/data_objects/Ed25519Key.java]: String.format()

File: app/src/test/java/com/nervousfish/nervousfish/data_objects/Ed25519KeyTest.java [hunk 3]
```diff
@@ -138,7 +139,7 @@ public void testSerialization() throws IOException, ClassNotFoundException {
                  ObjectInputStream ois = new ObjectInputStream(bis)) {
                 IKey key1 = (IKey) ois.readObject();
                 assertTrue(key1.getName().equals("foo"));
-                assertTrue(key1.getKey().equals("bar"));
+                assertTrue(key1.getKey().equals("bar" + "#" + Arrays.toString("foobar".getBytes())));
```
Comment (TheBonheurs) [inline app/src/test/java/com/nervousfish/nervousfish/data_objects/Ed25519KeyTest.java:137]: String.format()

Comment (clenrock) [inline app/src/test/java/com/nervousfish/nervousfish/data_objects/Ed25519KeyTest.java]: I think this is not needed for integration, with 1 day left to the deadline

Comment (clenrock) [reply app/src/test/java/com/nervousfish/nervousfish/data_objects/Ed25519KeyTest.java:137]: I think this is not needed for integration, with 1 day left to the deadline

**Pull request merged**